tools/content-generator/src/test/java/org/projectnessie/tools/contentgenerator/ITReadReferences.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.contentgenerator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.tools.contentgenerator.cli.NessieContentGenerator;
+
+class ITReadReferences extends AbstractContentGeneratorTest {
+
+  @Test
+  void readReferences() throws UnsupportedEncodingException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    PrintStream out = new PrintStream(baos);
+    System.setOut(out);
+
+    assertThat(NessieContentGenerator.runMain(new String[] {"refs", "--uri", NESSIE_API_URI}))
+        .isEqualTo(0);
+
+    out.close();
+    String[] output = baos.toString(StandardCharsets.UTF_8.toString()).split("\n");
```

Comment:
this pattern for capturing output looks bad...

the `CommandLine` class in `runMain` has methods `setOut` and `setErr` which seem like the right way to capture output in tests, wdyt?

if this does not work, then at least we should have a helper method (in base class) that does all this capturing ceremony.
as is we duplicate this boilerplate in 4 places now.